site/get-started.markdown

```diff
@@ -0,0 +1,143 @@
+
+## Set up Haskell dev environment
+
+### GHCup - universal installer
+
+[GHCup](https://www.haskell.org/ghcup/#) is a universal installer for Haskell that will install for you everything you need to program in Haskell, and then will also help you manage those installations in the future (update, switch versions, ...).
+
+It is simple to use, works the same way on all platforms (Linux, macOS, Windows), and gives you one central place/method to take care of your Haskell development setup.
+
+Follow instructions at [GHCup webpage](https://www.haskell.org/ghcup/#) to install GHCup. Then, use it to install the Haskell Toolchain, which consists of:
+
+1. **GHC** -> Haskell compiler. We will use it below to run our examples, but in practice, you will mostly be using a build tool like `cabal` or `Stack` to build your code, instead of `GHC` directly.
+2. **HLS** -> Haskell Language Server -> You won't use this directly, instead your code editor will use it in the background to provide you with a great experience while editing Haskell code.
+3. **cabal** -> Haskell build tool -> You will use this to structure your Haskell projects, build them, run them, define dependencies, ...
+4. **Stack** -> Haskell build tool -> alternative to `cabal`
+
+
+<div class="bs-callout bs-callout-info">
+  <p>
+    <h4>cabal and Stack -> which one should I install?</h4>
+    cabal is the original build tool, while Stack was created as an alternative to cabal a time ago to solve some of the problems that cabal had and provide a more user-friendly experience. In the meantime, cabal solved most of those issues (including "cabal hell") and caught up with Stack regarding user experience, so the main difference between them at the moment is how they resolve dependencies, which for you as a beginner isn't a concern. Therefore, both are a good choice, and if not sure, you can install both and then use whatever the learning resources you will later use will point you to.
+  </p>
+</div>
+
+To confirm it is all installed correctly, you can run the following commands:
+```
+> ghc --version
+> haskell-language-server-wrapper --version
+> cabal --version                               # If you installed cabal
+> stack --version                               # If you installed Stack
+```
+
+### Editor
+HLS (Haskell Language Server), which you just installed via GHCup, is the one that brings all the cool IDE features to editors, so as long as your editor has a decent Haskell language extension that utilizes HLS, you are good.
+
+The most straightforward option is to go with **Visual Studio Code** - just install [Haskell extension](https://marketplace.visualstudio.com/items?itemName=haskell.haskell) and you are all set, it should work out of the box and use your installation of HLS.
+
+Of other editors that have good Haskell extensions, the most popular ones are Vim and Emacs.
```

Comment:
I think this is redundant unless the we link to the editor integration plugins/methods - I think users of vim and emacs like to tweak with their editor and will likely try to find information on haskell editor integration with their editor of choice on their own.

Reply:
This is the getting started page, either you help me get started or you're just teasing :)

Reply:
I don't understand why wouldn't I be helping / how am I teasing if I mention vim and emacs? VSCode, vim and emacs are three most popular editors for Haskell, with good support all of them.

We highlight VSCode because it is the easiest to get started with, it is very popular, and setup is super easy. People who have VSCode already or don't use vim/emacs will follow this happily.

We mention vim and emacs shortly. If you are vim or emacs user, you will be happy to see this, and will investigate further. If you are not vim or emacs user, you will ignore this.

If needed I can remove this line, this is not something I care that much about to compromise the PR for it, but it feels a bit biased to me to mention only vscode.

Reply:
> I don't understand why wouldn't I be helping / how am I teasing if I mention vim and emacs?

What I'm trying to say is that emacs users are going to look for Haskell integration with emacs regardless to whether we list it here or not, because as an emacs user I do all of my programming in emacs regardless of the language. So I find it redundant, *unless* we help them get started. One way to help is to point to the relevant headings in the [hls configuration docs](https://haskell-language-server.readthedocs.io/en/latest/configuration.html).

What I'm trying to say is that the getting started page should not send anyone to google anything. If it's mentioned here, it should include enough context, or include the relevant links to more information. So either we don't write "emacs integration is available" and count on emacs users to google anyway (because they likely will), or we save them the search and send them to the relevant links.

Reply:
Sounds good @soupi , what I did now is removed the paragraph that mentions emacs and vim and replaced it with
```
To learn about support for other editors, check out [HLS docs for editor configuration](https://haskell-language-server.readthedocs.io/en/latest/configuration.html#configuring-your-editor).
```

Finally, if that still doesn't feel right, I can also remove that, but I think it now follows your recommendation from above.